packages/slate-editor/src/lib/scrollIntoView.ts

```diff
@@ -44,11 +44,8 @@ export function scrollIntoView(
         return;
     }
 
-    if (isChildAboveVisibleArea) {
+    if (isChildAboveVisibleArea || isChildBelowVisibleArea) {
         const y = parent.scrollTop + elementTop - minTop;
         scrollTo(parent, parent.scrollLeft, y);
-    } else if (isChildBelowVisibleArea) {
-        const y = parent.scrollTop + elementTop + elementHeight - parentHeight + minBottom;
-        scrollTo(parent, parent.scrollLeft, y);
```

Comment:
I still think the changed behavior is a step back from what it was before.

Scenarios:

1. I click on an image that is partially visible from below the viewport.

   The content jumps so that the image top edge is aligned with the viewport top edge.
   It also often happens that after I release the mouse button, it selects a portion of text. I believe this happens because I release the button outside of the clicked element, so it's treated as a range selection.

![Peek 2022-04-27 14-33](https://user-images.githubusercontent.com/370680/165509802-f54c6f43-5ea7-4f6e-98d2-c280cbe4bcbf.gif)

2. I use the keyboard `Down` button to navigate through content. 

    Normal editor behavior is to reveal content from below the viewport as I navigate, scrolling just enough to display the focused portion. When I navigate through a paragraph, it shows the next line at the bottom of the viewport. It doesn't scroll the viewport to put the focused line at the top.

   In contrast, when I focus a block element (like a header, divider, or an image) with a keyboard, it jumps to put the element at the top edge of the editor. This makes me lose orientation. It's not clear what happens, as suddenly the screen contains whole new content (there is no visual connection to any of the previously visible content).

![Peek 2022-04-27 14-34](https://user-images.githubusercontent.com/370680/165510351-9f14011d-bab1-47f2-bc56-aa993ab3d985.gif)